packages/core/src/schema/blocks/playground.ts

```diff
@@ -0,0 +1,204 @@
+import { Fragment, Schema } from "prosemirror-model";
+import { ViewMutationRecord } from "prosemirror-view";
+import { Block } from "../../blocks/defaultBlocks.js";
+import { BlockNoteEditor } from "../../editor/BlockNoteEditor.js";
+import { BlockNoteExtension } from "../../editor/BlockNoteExtension.js";
+import type { Props, PropSchema } from "../../schema/index.js";
+
+export type BlockDefs = Record<string, BlockConfig<string, PropSchema>>;
+
+export interface BlockConfigMeta {
+  /**
+   * Whether the block is selectable
+   */
+  selectable?: boolean;
+
+  /**
+   * The accept mime types for the file block
+   */
+  fileBlockAccept?: string[];
+
+  /**
+   * Whether the block is a {@link https://prosemirror.net/docs/ref/#model.NodeSpec.code} block
+   */
+  code?: boolean;
+
+  /**
+   * Whether the block is a {@link https://prosemirror.net/docs/ref/#model.NodeSpec.defining} block
+   */
+  defining?: boolean;
+}
+
+/**
+ * BlockConfig contains the "schema" info about a Block type
+ * i.e. what props it supports, what content it supports, etc.
+ */
+export interface BlockConfig<
+  TName extends string,
+  TSchema extends PropSchema = PropSchema,
+> {
+  /**
+   * The type of the block (unique identifier within a schema)
+   */
+  type: TName;
+  /**
+   * The properties that the block supports
+   * @todo will be zod schema in the future
+   */
+  readonly propSchema: TSchema;
+  /**
+   * The content that the block supports
+   */
+  content: "inline" | "none";
+  // TODO: how do you represent things that have nested content?
+  // e.g. tables, alerts (with title & content)
+  /**
+   * Metadata
+   */
+  meta?: BlockConfigMeta;
+}
+
+export interface BlockImplementation<
+  TName extends string,
+  TProps extends PropSchema,
+> {
+  /**
+   * A function that converts the block into a DOM element
+   */
+  render: (
+    /**
+     * The custom block to render
+     */
+    block: Block<Record<TName, BlockConfig<TName, TProps>>>,
+    /**
+     * The BlockNote editor instance
+     */
+    editor: BlockNoteEditor<Record<TName, BlockConfig<TName, TProps>>>,
+  ) => {
```

Comment:
These functions (dom / contentDOM / ignoreMutation) are very prosemirror specific. For now out of scope (as they're also in the existing API and this PR is about moving blocks over), but don't think this is ideal

Reply:
Yep, agreed, but not looking to rock that boat right now